hassio/src/components/supervisor-snapshot-content.ts

```diff
@@ -0,0 +1,362 @@
+
+  static get styles(): CSSResultGroup {
+    return css`
+      ha-checkbox {
+        --mdc-checkbox-touch-target-size: 16px;
+        display: block;
+        margin: 4px 12px 8px 0;
+      }
+      ha-formfield {
+        display: block;
+      }
+      supervisor-formfield-label {
+        display: inline-flex;
+        align-items: center;
+      }
+      paper-input[type="password"] {
+        display: block;
+        margin: 4px 0 4px 16px;
+      }
+      .details {
+        color: var(--secondary-text-color);
+      }
+      .section-content {
+        display: flex;
+        flex-direction: column;
+        margin-left: 16px;
+      }
+      .security {
+        margin-top: 16px;
+      }
+    `;
+  }
+
+  private _getSection(section: string) {
+    const templates: TemplateResult[] = [];
+    let checkedItems = 0;
+    this[section].forEach((item) => {
+      templates.push(html`<ha-formfield
+        .label=${html`<supervisor-formfield-label
+          .label=${item.name}
+          .iconPath=${section === "folders" ? mdiFolder : mdiPuzzle}
+          .imageUrl=${section === "folders"
+            ? undefined
+            : atLeastVersion(this.hass.config.version, 0, 105) &&
+              this.supervisor!.addon.addons.find(
+                (addon) => addon.slug === item.slug
+              )?.icon
+            ? `/api/hassio/addons/${item.slug}/icon`
+            : undefined}
+          .version=${item.version}
+        >
+        </supervisor-formfield-label>`}
+      >
+        <ha-checkbox
+          .item=${item}
+          .checked=${item.checked}
+          .section=${section}
+          @change=${this._updateSectionEntry}
+        >
+        </ha-checkbox>
+      </ha-formfield>`);
+
+      if (item.checked) {
+        checkedItems++;
+      }
+    });
+
+    const checked = checkedItems === this[section].length;
+
+    return {
+      templates,
+      checked,
+      indeterminate: !checked && checkedItems !== 0,
+    };
+  }
+
+  private _handleRadioValueChanged(ev: CustomEvent) {
+    const input = ev.currentTarget as HaRadio;
+    this[input.name] = input.value;
+  }
+
+  private _handleTextValueChanged(ev: PolymerChangedEvent<string>) {
+    const input = ev.currentTarget as PaperInputElement;
+    this[input.name!] = ev.detail.value;
+  }
+
+  private _toggleHasPassword(): void {
+    this.snapshotHasPassword = !this.snapshotHasPassword;
+  }
+
+  private _toggleSection(ev): void {
+    const section = ev.currentTarget.section;
+
+    this[section] = (section === "addons" ? this.addons : this.folders)!.map(
+      (item) => ({
+        ...item,
+        checked: ev.currentTarget.checked,
+      })
+    );
+  }
+
+  private _updateSectionEntry(ev): void {
+    const item = ev.currentTarget.item;
+    const section = ev.currentTarget.section;
+    this[section] = this[section].map((entry) => ({
+      ...entry,
+      checked:
+        entry.slug === item.slug ? ev.currentTarget.checked : entry.checked,
+    }));
+  }
```

Comment:
We now make a copy of every item when 1 item changes.
```suggestion
    this[section] = this[section].map((entry) => entry.slug === item.slug ? {
      ...entry,
      checked: ev.currentTarget.checked,
    } : entry);
  }
```